### PR TITLE
Fix EZP-23978: Don't forward on index request when in legacy_mode

### PR DESCRIPTION
**Pull request by joaoinacio**
JIRA: https://jira.ez.no/browse/EZP-23978

Override IndexRequest listener, do not forward request when in legacy mode.

Depends on https://github.com/ezsystems/ezpublish-kernel/pull/1187

### CODE REVIEW

File: bundle/EventListener/IndexRequestListener.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Routing\RouterInterface;
+
+class IndexRequestListener extends CoreIndexListener
+{
+    /**
+     * @var \eZ\Publish\Core\MVC\ConfigResolverInterface
+     */
+    private $configResolver;
```
Comment (lolautruche) [inline bundle/EventListener/IndexRequestListener.php]: Property is already present in parent class, just mark it protected there.

File: bundle/EventListener/IndexRequestListener.php [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    public function onKernelRequestIndex( GetResponseEvent $event )
+    {
+        $isLegacyMode = $this->configResolver->getParameter( 'legacy_mode' );
+        if ( $isLegacyMode )
+        {
+            return;
+        }
+        parent::onKernelRequestIndex( $event );
+    }
+
```
Comment (lolautruche) [inline bundle/EventListener/IndexRequestListener.php]: CS: Extra empty line

File: bundle/EventListener/IndexRequestListener.php [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    /**
+     * Overrides core index request, which checks if the IndexPage is configured and which page must be shown.
+     * If matched SiteAccess uses legacy mode, do not execute event.
+     *
+     * @param GetResponseEvent $event
+     */
+    public function onKernelRequestIndex( GetResponseEvent $event )
+    {
+        $isLegacyMode = $this->configResolver->getParameter( 'legacy_mode' );
```
Comment (lolautruche) [inline bundle/EventListener/IndexRequestListener.php]: Temp variable not needed

File: bundle/Resources/config/services.yml [hunk 4]
```diff
@@ -295,6 +296,17 @@ services:
...
+    ezpublish.request_index_listener:
+        class: %ezpublish_legacy.index_request_listener.class%
+        arguments:
+            - @ezpublish.config.resolver
+            - @router
+            - %ezpublish.siteaccess.default%
+            - @?logger
+        tags:
+            - { name: kernel.event_subscriber }
+
```
Comment (lolautruche) [inline bundle/Resources/config/services.yml]: Why do you create a new service? All you have to do is to override `ezpublish.request_index_listener.class` parameter

Comment (joaoinacio) [inline bundle/Resources/config/services.yml]: got it, thanks!

**Comment by joaoinacio** [comment]
Updated, does this need any particular testing?

If so, I am guessing a new PR is needed against community unless https://github.com/ezsystems/ezpublish-kernel/pull/1187 is merged first

File: bundle/EventListener/IndexRequestListener.php [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+namespace eZ\Bundle\EzPublishLegacyBundle\EventListener;
+
+use eZ\Bundle\EzPublishCoreBundle\EventListener\IndexRequestListener as CoreIndexListener;
+use eZ\Publish\Core\MVC\ConfigResolverInterface;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+use Symfony\Component\DependencyInjection\ContainerAware;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Psr\Log\LoggerInterface;
```
Comment (yannickroger) [inline bundle/EventListener/IndexRequestListener.php]:
```
use eZ\Publish\Core\MVC\ConfigResolverInterface;
use Symfony\Component\HttpKernel\KernelEvents;
use Symfony\Component\DependencyInjection\ContainerAware;
use Symfony\Component\EventDispatcher\EventSubscriberInterface;
use Psr\Log\LoggerInterface;
use Symfony\Component\Routing\RouterInterface;
```

**Comment by yannickroger** [comment]
+1 with remarks
I think we should merge https://github.com/ezsystems/ezpublish-kernel/pull/1187 first.

**Comment by yannickroger** [comment]
@joaoinacio if you are bored you can start working on the backport as it will be quite specific :)

**Comment by andrerom** [comment]
+0,9 review ping @lolautruche / @bdunogier

File: bundle/Resources/config/services.yml [hunk 6]
```diff
@@ -70,6 +70,9 @@ parameters:
     ezpublish_legacy.config_scope_listener.class: eZ\Bundle\EzPublishLegacyBundle\EventListener\ConfigScopeListener
     ezpublish_legacy.legacy_kernel_listener.class: eZ\Bundle\EzPublishLegacyBundle\EventListener\LegacyKernelListener
 
+        # Overriding base IndexRequest event listener
```
Comment (lolautruche) [inline bundle/Resources/config/services.yml]: Extra WS

**Comment by lolautruche** [comment]
+1

**Comment by lolautruche** [comment]
@joaoinacio You can merge this one as soon as https://github.com/ezsystems/ezpublish-kernel/pull/1187 is merged.

**Pull request merged**